src/test/unit_tests/jest_tests/test_nc_upgrade_manager.test.js

```diff
@@ -673,6 +726,101 @@ describe('nc upgrade manager - upgrade config directory', () => {
+            await create_config_dir(config_root);
+            await create_redirect_file(config_fs, config_root);
+        });
+
+        afterEach(async () => {
+            await clean_config_dir(config_fs, config_root);
+        }, TEST_TIMEOUT);
+
+        it('upgrade_config_dir - empty system_data', async () => {
+            const expected_version = pkg.version;
+            const hosts_list = [hostname];
+            const expected_err_msg = 'system does not exist';
+            await expect(nc_upgrade_manager.upgrade_config_dir(expected_version, hosts_list))
+                .rejects.toThrow(expected_err_msg);
+        });
+
+        it('upgrade_config_dir - config_directory property is missing in system.json', async () => {
+            const system_data = _.cloneDeep(current_expected_system_json);
+            const expected_version = pkg.version;
+            const hosts_list = [hostname];
+            await config_fs.create_system_config_file(JSON.stringify(system_data));
+            const res = await nc_upgrade_manager.upgrade_config_dir(expected_version, hosts_list);
+            const system_json_data = await config_fs.get_system_config_file();
+            const expected_data = {
+                ...system_data,
+                config_directory: {
+                    phase: CONFIG_DIR_PHASES.CONFIG_DIR_UNLOCKED,
+                    config_dir_version: config_fs.config_dir_version,
+                    upgrade_package_version: expected_version,
+                    upgrade_history: {
+                        successful_upgrades: [res]
+                    }
+                }
+            };
+            assert_upgrade_finish_or_fail_data(system_json_data, expected_data);
+        });
+
+        it('upgrade_config_dir - system.json and package.json have the same version', async () => {
+            const system_data = _.cloneDeep(old_expected_system_json_has_config_directory);
+            const expected_version = pkg.version;
+            const hosts_list = [hostname];
+            await config_fs.create_system_config_file(JSON.stringify(system_data));
+            const expected_msg = { "message": "config_dir_version on system.json and config_fs.config_dir_version match, nothing to upgrade" };
+            const res = await nc_upgrade_manager.upgrade_config_dir(expected_version, hosts_list);
+            expect(res).toStrictEqual(expected_msg);
+        });
+
+        it('upgrade_config_dir - config_dir_version in system.json is newer than the hosts current config_dir_version', async () => {
+            const system_data = _.cloneDeep(newer_expected_system_json_has_config_directory);
+            const expected_version = pkg.version;
+            const hosts_list = [hostname];
+            await config_fs.create_system_config_file(JSON.stringify(system_data));
+            const expected_err_msg = 'attempt to run old container version with newer server version';
```

Comment:
container? Do we use this term in NC?

Reply:
currently yes because it's in the utils, but I'll change it in a following PR